---
title: "Griffin R Assignment"
output:
  pdf_document: default
  html_document: default
---

### R Markdown

##Data Inspection 

#what type of file?
setwd("~/Desktop/R_Assignment")

``` {r}
typeof(fang_et_al_genotypes) 
```
[1] "list"

```{r}
typeof(snp_position)
```
[1] "list"
#How long?
``` {r}
length(fang_et_al_genotypes)
```
[1] 986
```{r}
length(snp_position)
```
[1] 15

#What attributes do the files have?
```{r}
attributes(fang_et_al_genotypes)
```
$names
  [1] "V1"   "V2"   "V3"   "V4"   "V5"   "V6"   "V7"   "V8"   "V9"   "V10"  "V11"  "V12"  "V13"  "V14"  "V15"  "V16"  "V17"  "V18"  "V19"  "V20" 
 [21] "V21"  "V22"  "V23"  "V24"  "V25"  "V26"  "V27"  "V28"  "V29"  "V30"  "V31"  "V32"  "V33"  "V34"  "V35"  "V36"  "V37"  "V38"  "V39"  "V40" 
 [41] "V41"  "V42"  "V43"  "V44"  "V45"  "V46"  "V47"  "V48"  "V49"  "V50"  "V51"  "V52"  "V53"  "V54"  "V55"  "V56"  "V57"  "V58"  "V59"  "V60" 
 [61] "V61"  "V62"  "V63"  "V64"  "V65"  "V66"  "V67"  "V68"  "V69"  "V70"  "V71"  "V72"  "V73"  "V74"  "V75"  "V76"  "V77"  "V78"  "V79"  "V80" 
 [81] "V81"  "V82"  "V83"  "V84"  "V85"  "V86"  "V87"  "V88"  "V89"  "V90"  "V91"  "V92"  "V93"  "V94"  "V95"  "V96"  "V97"  "V98"  "V99"  "V100"
[101] "V101" "V102" "V103" "V104" "V105" "V106" "V107" "V108" "V109" "V110" "V111" "V112" "V113" "V114" "V115" "V116" "V117" "V118" "V119" "V120"
[121] "V121" "V122" "V123" "V124" "V125" "V126" "V127" "V128" "V129" "V130" "V131" "V132" "V133" "V134" "V135" "V136" "V137" "V138" "V139" "V140"
[141] "V141" "V142" "V143" "V144" "V145" "V146" "V147" "V148" "V149" "V150" "V151" "V152" "V153" "V154" "V155" "V156" "V157" "V158" "V159" "V160"
[161] "V161" "V162" "V163" "V164" "V165" "V166" "V167" "V168" "V169" "V170" "V171" "V172" "V173" "V174" "V175" "V176" "V177" "V178" "V179" "V180"
[181] "V181" "V182" "V183" "V184" "V185" "V186" "V187" "V188" "V189" "V190" "V191" "V192" "V193" "V194" "V195" "V196" "V197" "V198" "V199" "V200"
[201] "V201" "V202" "V203" "V204" "V205" "V206" "V207" "V208" "V209" "V210" "V211" "V212" "V213" "V214" "V215" "V216" "V217" "V218" "V219" "V220"
[221] "V221" "V222" "V223" "V224" "V225" "V226" "V227" "V228" "V229" "V230" "V231" "V232" "V233" "V234" "V235" "V236" "V237" "V238" "V239" "V240"
[241] "V241" "V242" "V243" "V244" "V245" "V246" "V247" "V248" "V249" "V250" "V251" "V252" "V253" "V254" "V255" "V256" "V257" "V258" "V259" "V260"
[261] "V261" "V262" "V263" "V264" "V265" "V266" "V267" "V268" "V269" "V270" "V271" "V272" "V273" "V274" "V275" "V276" "V277" "V278" "V279" "V280"
[281] "V281" "V282" "V283" "V284" "V285" "V286" "V287" "V288" "V289" "V290" "V291" "V292" "V293" "V294" "V295" "V296" "V297" "V298" "V299" "V300"
[301] "V301" "V302" "V303" "V304" "V305" "V306" "V307" "V308" "V309" "V310" "V311" "V312" "V313" "V314" "V315" "V316" "V317" "V318" "V319" "V320"
[321] "V321" "V322" "V323" "V324" "V325" "V326" "V327" "V328" "V329" "V330" "V331" "V332" "V333" "V334" "V335" "V336" "V337" "V338" "V339" "V340"
[341] "V341" "V342" "V343" "V344" "V345" "V346" "V347" "V348" "V349" "V350" "V351" "V352" "V353" "V354" "V355" "V356" "V357" "V358" "V359" "V360"
[361] "V361" "V362" "V363" "V364" "V365" "V366" "V367" "V368" "V369" "V370" "V371" "V372" "V373" "V374" "V375" "V376" "V377" "V378" "V379" "V380"
[381] "V381" "V382" "V383" "V384" "V385" "V386" "V387" "V388" "V389" "V390" "V391" "V392" "V393" "V394" "V395" "V396" "V397" "V398" "V399" "V400"
[401] "V401" "V402" "V403" "V404" "V405" "V406" "V407" "V408" "V409" "V410" "V411" "V412" "V413" "V414" "V415" "V416" "V417" "V418" "V419" "V420"
[421] "V421" "V422" "V423" "V424" "V425" "V426" "V427" "V428" "V429" "V430" "V431" "V432" "V433" "V434" "V435" "V436" "V437" "V438" "V439" "V440"
[441] "V441" "V442" "V443" "V444" "V445" "V446" "V447" "V448" "V449" "V450" "V451" "V452" "V453" "V454" "V455" "V456" "V457" "V458" "V459" "V460"
[461] "V461" "V462" "V463" "V464" "V465" "V466" "V467" "V468" "V469" "V470" "V471" "V472" "V473" "V474" "V475" "V476" "V477" "V478" "V479" "V480"
[481] "V481" "V482" "V483" "V484" "V485" "V486" "V487" "V488" "V489" "V490" "V491" "V492" "V493" "V494" "V495" "V496" "V497" "V498" "V499" "V500"
[501] "V501" "V502" "V503" "V504" "V505" "V506" "V507" "V508" "V509" "V510" "V511" "V512" "V513" "V514" "V515" "V516" "V517" "V518" "V519" "V520"
[521] "V521" "V522" "V523" "V524" "V525" "V526" "V527" "V528" "V529" "V530" "V531" "V532" "V533" "V534" "V535" "V536" "V537" "V538" "V539" "V540"
[541] "V541" "V542" "V543" "V544" "V545" "V546" "V547" "V548" "V549" "V550" "V551" "V552" "V553" "V554" "V555" "V556" "V557" "V558" "V559" "V560"
[561] "V561" "V562" "V563" "V564" "V565" "V566" "V567" "V568" "V569" "V570" "V571" "V572" "V573" "V574" "V575" "V576" "V577" "V578" "V579" "V580"
[581] "V581" "V582" "V583" "V584" "V585" "V586" "V587" "V588" "V589" "V590" "V591" "V592" "V593" "V594" "V595" "V596" "V597" "V598" "V599" "V600"
[601] "V601" "V602" "V603" "V604" "V605" "V606" "V607" "V608" "V609" "V610" "V611" "V612" "V613" "V614" "V615" "V616" "V617" "V618" "V619" "V620"
[621] "V621" "V622" "V623" "V624" "V625" "V626" "V627" "V628" "V629" "V630" "V631" "V632" "V633" "V634" "V635" "V636" "V637" "V638" "V639" "V640"
[641] "V641" "V642" "V643" "V644" "V645" "V646" "V647" "V648" "V649" "V650" "V651" "V652" "V653" "V654" "V655" "V656" "V657" "V658" "V659" "V660"
[661] "V661" "V662" "V663" "V664" "V665" "V666" "V667" "V668" "V669" "V670" "V671" "V672" "V673" "V674" "V675" "V676" "V677" "V678" "V679" "V680"
[681] "V681" "V682" "V683" "V684" "V685" "V686" "V687" "V688" "V689" "V690" "V691" "V692" "V693" "V694" "V695" "V696" "V697" "V698" "V699" "V700"
[701] "V701" "V702" "V703" "V704" "V705" "V706" "V707" "V708" "V709" "V710" "V711" "V712" "V713" "V714" "V715" "V716" "V717" "V718" "V719" "V720"
[721] "V721" "V722" "V723" "V724" "V725" "V726" "V727" "V728" "V729" "V730" "V731" "V732" "V733" "V734" "V735" "V736" "V737" "V738" "V739" "V740"
[741] "V741" "V742" "V743" "V744" "V745" "V746" "V747" "V748" "V749" "V750" "V751" "V752" "V753" "V754" "V755" "V756" "V757" "V758" "V759" "V760"
[761] "V761" "V762" "V763" "V764" "V765" "V766" "V767" "V768" "V769" "V770" "V771" "V772" "V773" "V774" "V775" "V776" "V777" "V778" "V779" "V780"
[781] "V781" "V782" "V783" "V784" "V785" "V786" "V787" "V788" "V789" "V790" "V791" "V792" "V793" "V794" "V795" "V796" "V797" "V798" "V799" "V800"
[801] "V801" "V802" "V803" "V804" "V805" "V806" "V807" "V808" "V809" "V810" "V811" "V812" "V813" "V814" "V815" "V816" "V817" "V818" "V819" "V820"
[821] "V821" "V822" "V823" "V824" "V825" "V826" "V827" "V828" "V829" "V830" "V831" "V832" "V833" "V834" "V835" "V836" "V837" "V838" "V839" "V840"
[841] "V841" "V842" "V843" "V844" "V845" "V846" "V847" "V848" "V849" "V850" "V851" "V852" "V853" "V854" "V855" "V856" "V857" "V858" "V859" "V860"
[861] "V861" "V862" "V863" "V864" "V865" "V866" "V867" "V868" "V869" "V870" "V871" "V872" "V873" "V874" "V875" "V876" "V877" "V878" "V879" "V880"
[881] "V881" "V882" "V883" "V884" "V885" "V886" "V887" "V888" "V889" "V890" "V891" "V892" "V893" "V894" "V895" "V896" "V897" "V898" "V899" "V900"
[901] "V901" "V902" "V903" "V904" "V905" "V906" "V907" "V908" "V909" "V910" "V911" "V912" "V913" "V914" "V915" "V916" "V917" "V918" "V919" "V920"
[921] "V921" "V922" "V923" "V924" "V925" "V926" "V927" "V928" "V929" "V930" "V931" "V932" "V933" "V934" "V935" "V936" "V937" "V938" "V939" "V940"
[941] "V941" "V942" "V943" "V944" "V945" "V946" "V947" "V948" "V949" "V950" "V951" "V952" "V953" "V954" "V955" "V956" "V957" "V958" "V959" "V960"
[961] "V961" "V962" "V963" "V964" "V965" "V966" "V967" "V968" "V969" "V970" "V971" "V972" "V973" "V974" "V975" "V976" "V977" "V978" "V979" "V980"
[981] "V981" "V982" "V983" "V984" "V985" "V986"

$class
[1] "data.frame"

$row.names
   [1]    1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16   17   18   19   20   21   22   23   24   25   26   27   28
  [29]   29   30   31   32   33   34   35   36   37   38   39   40   41   42   43   44   45   46   47   48   49   50   51   52   53   54   55   56
  [57]   57   58   59   60   61   62   63   64   65   66   67   68   69   70   71   72   73   74   75   76   77   78   79   80   81   82   83   84
  [85]   85   86   87   88   89   90   91   92   93   94   95   96   97   98   99  100  101  102  103  104  105  106  107  108  109  110  111  112
 [113]  113  114  115  116  117  118  119  120  121  122  123  124  125  126  127  128  129  130  131  132  133  134  135  136  137  138  139  140
 [141]  141  142  143  144  145  146  147  148  149  150  151  152  153  154  155  156  157  158  159  160  161  162  163  164  165  166  167  168
 [169]  169  170  171  172  173  174  175  176  177  178  179  180  181  182  183  184  185  186  187  188  189  190  191  192  193  194  195  196
 [197]  197  198  199  200  201  202  203  204  205  206  207  208  209  210  211  212  213  214  215  216  217  218  219  220  221  222  223  224
 [225]  225  226  227  228  229  230  231  232  233  234  235  236  237  238  239  240  241  242  243  244  245  246  247  248  249  250  251  252
 [253]  253  254  255  256  257  258  259  260  261  262  263  264  265  266  267  268  269  270  271  272  273  274  275  276  277  278  279  280
 [281]  281  282  283  284  285  286  287  288  289  290  291  292  293  294  295  296  297  298  299  300  301  302  303  304  305  306  307  308
 [309]  309  310  311  312  313  314  315  316  317  318  319  320  321  322  323  324  325  326  327  328  329  330  331  332  333  334  335  336
 [337]  337  338  339  340  341  342  343  344  345  346  347  348  349  350  351  352  353  354  355  356  357  358  359  360  361  362  363  364
 [365]  365  366  367  368  369  370  371  372  373  374  375  376  377  378  379  380  381  382  383  384  385  386  387  388  389  390  391  392
 [393]  393  394  395  396  397  398  399  400  401  402  403  404  405  406  407  408  409  410  411  412  413  414  415  416  417  418  419  420
 [421]  421  422  423  424  425  426  427  428  429  430  431  432  433  434  435  436  437  438  439  440  441  442  443  444  445  446  447  448
 [449]  449  450  451  452  453  454  455  456  457  458  459  460  461  462  463  464  465  466  467  468  469  470  471  472  473  474  475  476
 [477]  477  478  479  480  481  482  483  484  485  486  487  488  489  490  491  492  493  494  495  496  497  498  499  500  501  502  503  504
 [505]  505  506  507  508  509  510  511  512  513  514  515  516  517  518  519  520  521  522  523  524  525  526  527  528  529  530  531  532
 [533]  533  534  535  536  537  538  539  540  541  542  543  544  545  546  547  548  549  550  551  552  553  554  555  556  557  558  559  560
 [561]  561  562  563  564  565  566  567  568  569  570  571  572  573  574  575  576  577  578  579  580  581  582  583  584  585  586  587  588
 [589]  589  590  591  592  593  594  595  596  597  598  599  600  601  602  603  604  605  606  607  608  609  610  611  612  613  614  615  616
 [617]  617  618  619  620  621  622  623  624  625  626  627  628  629  630  631  632  633  634  635  636  637  638  639  640  641  642  643  644
 [645]  645  646  647  648  649  650  651  652  653  654  655  656  657  658  659  660  661  662  663  664  665  666  667  668  669  670  671  672
 [673]  673  674  675  676  677  678  679  680  681  682  683  684  685  686  687  688  689  690  691  692  693  694  695  696  697  698  699  700
 [701]  701  702  703  704  705  706  707  708  709  710  711  712  713  714  715  716  717  718  719  720  721  722  723  724  725  726  727  728
 [729]  729  730  731  732  733  734  735  736  737  738  739  740  741  742  743  744  745  746  747  748  749  750  751  752  753  754  755  756
 [757]  757  758  759  760  761  762  763  764  765  766  767  768  769  770  771  772  773  774  775  776  777  778  779  780  781  782  783  784
 [785]  785  786  787  788  789  790  791  792  793  794  795  796  797  798  799  800  801  802  803  804  805  806  807  808  809  810  811  812
 [813]  813  814  815  816  817  818  819  820  821  822  823  824  825  826  827  828  829  830  831  832  833  834  835  836  837  838  839  840
 [841]  841  842  843  844  845  846  847  848  849  850  851  852  853  854  855  856  857  858  859  860  861  862  863  864  865  866  867  868
 [869]  869  870  871  872  873  874  875  876  877  878  879  880  881  882  883  884  885  886  887  888  889  890  891  892  893  894  895  896
 [897]  897  898  899  900  901  902  903  904  905  906  907  908  909  910  911  912  913  914  915  916  917  918  919  920  921  922  923  924
 [925]  925  926  927  928  929  930  931  932  933  934  935  936  937  938  939  940  941  942  943  944  945  946  947  948  949  950  951  952
 [953]  953  954  955  956  957  958  959  960  961  962  963  964  965  966  967  968  969  970  971  972  973  974  975  976  977  978  979  980
 [981]  981  982  983  984  985  986  987  988  989  990  991  992  993  994  995  996  997  998  999 1000
 [ reached getOption("max.print") -- omitted 1783 entries ]

``` {r}
attributes(snp_position)
```
$names
 [1] "SNP_ID"               "cdv_marker_id"        "Chromosome"           "Position"             "alt_pos"              "mult_positions"      
 [7] "amplicon"             "cdv_map_feature.name" "gene"                 "candidate.random"     "Genaissance_daa_id"   "Sequenom_daa_id"     
[13] "count_amplicons"      "count_cmf"            "count_gene"          

$class
[1] "data.frame"

$row.names
  [1]   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35
 [36]  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70
 [71]  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105
[106] 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140
[141] 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175
[176] 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210
[211] 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245
[246] 246 247 248 249 250 251 252 253 254 255 256 257 258 259 260 261 262 263 264 265 266 267 268 269 270 271 272 273 274 275 276 277 278 279 280
[281] 281 282 283 284 285 286 287 288 289 290 291 292 293 294 295 296 297 298 299 300 301 302 303 304 305 306 307 308 309 310 311 312 313 314 315
[316] 316 317 318 319 320 321 322 323 324 325 326 327 328 329 330 331 332 333 334 335 336 337 338 339 340 341 342 343 344 345 346 347 348 349 350
[351] 351 352 353 354 355 356 357 358 359 360 361 362 363 364 365 366 367 368 369 370 371 372 373 374 375 376 377 378 379 380 381 382 383 384 385
[386] 386 387 388 389 390 391 392 393 394 395 396 397 398 399 400 401 402 403 404 405 406 407 408 409 410 411 412 413 414 415 416 417 418 419 420
[421] 421 422 423 424 425 426 427 428 429 430 431 432 433 434 435 436 437 438 439 440 441 442 443 444 445 446 447 448 449 450 451 452 453 454 455
[456] 456 457 458 459 460 461 462 463 464 465 466 467 468 469 470 471 472 473 474 475 476 477 478 479 480 481 482 483 484 485 486 487 488 489 490
[491] 491 492 493 494 495 496 497 498 499 500 501 502 503 504 505 506 507 508 509 510 511 512 513 514 515 516 517 518 519 520 521 522 523 524 525
[526] 526 527 528 529 530 531 532 533 534 535 536 537 538 539 540 541 542 543 544 545 546 547 548 549 550 551 552 553 554 555 556 557 558 559 560
[561] 561 562 563 564 565 566 567 568 569 570 571 572 573 574 575 576 577 578 579 580 581 582 583 584 585 586 587 588 589 590 591 592 593 594 595
[596] 596 597 598 599 600 601 602 603 604 605 606 607 608 609 610 611 612 613 614 615 616 617 618 619 620 621 622 623 624 625 626 627 628 629 630
[631] 631 632 633 634 635 636 637 638 639 640 641 642 643 644 645 646 647 648 649 650 651 652 653 654 655 656 657 658 659 660 661 662 663 664 665
[666] 666 667 668 669 670 671 672 673 674 675 676 677 678 679 680 681 682 683 684 685 686 687 688 689 690 691 692 693 694 695 696 697 698 699 700
[701] 701 702 703 704 705 706 707 708 709 710 711 712 713 714 715 716 717 718 719 720 721 722 723 724 725 726 727 728 729 730 731 732 733 734 735
[736] 736 737 738 739 740 741 742 743 744 745 746 747 748 749 750 751 752 753 754 755 756 757 758 759 760 761 762 763 764 765 766 767 768 769 770
[771] 771 772 773 774 775 776 777 778 779 780 781 782 783 784 785 786 787 788 789 790 791 792 793 794 795 796 797 798 799 800 801 802 803 804 805
[806] 806 807 808 809 810 811 812 813 814 815 816 817 818 819 820 821 822 823 824 825 826 827 828 829 830 831 832 833 834 835 836 837 838 839 840
[841] 841 842 843 844 845 846 847 848 849 850 851 852 853 854 855 856 857 858 859 860 861 862 863 864 865 866 867 868 869 870 871 872 873 874 875
[876] 876 877 878 879 880 881 882 883 884 885 886 887 888 889 890 891 892 893 894 895 896 897 898 899 900 901 902 903 904 905 906 907 908 909 910
[911] 911 912 913 914 915 916 917 918 919 920 921 922 923 924 925 926 927 928 929 930 931 932 933 934 935 936 937 938 939 940 941 942 943 944 945
[946] 946 947 948 949 950 951 952 953 954 955 956 957 958 959 960 961 962 963 964 965 966 967 968 969 970 971 972 973 974 975 976 977 978 979 980
[981] 981 982 983

#Is the file recursive and/or numeric?
```{r}
is.recursive(fang_et_al_genotypes)
```
[1] TRUE
```{r}
is.recursive(snp_position)
```
[1] TRUE
```{r}
is.numeric(fang_et_al_genotypes)
```
[1] FALSE
``` {r}
is.numeric(snp_position)
```
[1] FALSE

#How many rows and coloumns?
``` {r}
nrow(fang_et_al_genotypes)
```
[1] 2783
```{r}
nrow(snp_position)
```
[1] 983
```{r}
ncol(fang_et_al_genotypes)
```
[1] 986
```{r}
ncol(snp_position)
```
[1] 15

#What is the structure of the data frame?
```{r}
str(fang_et_al_genotypes)
```
'data.frame':	2783 obs. of  986 variables:
 $ V1  : Factor w/ 2783 levels "BKN_001","BKN_002",..: 441 446 447 443 444 449 451 458 460 461 ...
 $ V2  : Factor w/ 2783 levels "JG_OTU","T-aust-1",..: 1 2 3 4 5 6 7 8 9 10 ...
 $ V3  : Factor w/ 17 levels "Group","TRIPS",..: 1 2 2 2 2 2 2 2 2 2 ...
 $ V4  : Factor w/ 5 levels "?/?","abph1.20",..: 2 1 1 1 1 1 1 1 1 1 ...
 $ V5  : Factor w/ 5 levels "?/?","A/A","A/C",..: 4 1 1 1 1 1 1 1 1 1 ...
 $ V6  : Factor w/ 5 levels "?/?","ae1.3",..: 2 5 5 5 5 5 5 5 5 5 ...
 $ V7  : Factor w/ 5 levels "?/?","A/A","A/G",..: 4 5 1 5 5 5 5 5 5 5 ...
 $ V8  : Factor w/ 5 levels "?/?","ae1.5",..: 2 5 5 5 5 5 5 5 5 5 ...
 $ V9  : Factor w/ 5 levels "?/?","an1.4",..: 2 3 3 1 3 3 3 3 3 3 ...
 $ V10 : Factor w/ 5 levels "?/?","A/A","A/G",..: 4 1 3 5 5 1 3 5 1 5 ...
 $ V11 : Factor w/ 5 levels "?/?","ba1.9",..: 2 3 3 3 3 3 3 3 3 3 ...
 $ V12 : Factor w/ 5 levels "?/?","bt2.5",..: 2 1 1 3 3 1 1 1 1 1 ...
 $ V13 : Factor w/ 5 levels "?/?","A/A","A/G",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V14 : Factor w/ 5 levels "?/?","A/A","A/G",..: 4 1 1 1 1 1 1 1 1 1 ...
 $ V15 : Factor w/ 5 levels "?/?","C/C","C/G",..: 4 2 2 1 1 2 2 2 1 2 ...
 $ V16 : Factor w/ 5 levels "?/?","A/A","A/G",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V17 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 5 5 5 5 5 5 5 5 5 ...
 $ V18 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V19 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 2 2 2 2 2 2 2 2 2 ...
 $ V20 : Factor w/ 5 levels "?/?","A/A","A/T",..: 4 1 5 5 5 5 5 5 5 5 ...
 $ V21 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 1 1 1 1 1 1 1 1 1 ...
 $ V22 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 1 1 1 1 1 1 1 1 1 ...
 $ V23 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 1 1 2 2 1 1 1 1 1 ...
 $ V24 : Factor w/ 5 levels "?/?","A/A","A/C",..: 5 1 1 1 1 1 1 1 1 1 ...
 $ V25 : Factor w/ 5 levels "?/?","A/A","A/C",..: 5 4 4 4 4 4 4 4 4 4 ...
 $ V26 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V27 : Factor w/ 5 levels "?/?","A/A","A/C",..: 5 1 1 4 1 4 4 4 4 4 ...
 $ V28 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 5 5 1 5 5 5 5 5 5 ...
 $ V29 : Factor w/ 5 levels "?/?","G/G","G/T",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V30 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V31 : Factor w/ 5 levels "?/?","A/A","A/T",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V32 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 1 4 4 4 4 4 4 4 4 ...
 $ V33 : Factor w/ 5 levels "?/?","C/C","C/G",..: 5 2 2 2 2 2 3 2 2 2 ...
 $ V34 : Factor w/ 5 levels "?/?","A/A","A/C",..: 5 4 4 4 4 1 1 4 4 4 ...
 $ V35 : Factor w/ 5 levels "?/?","A/A","A/T",..: 4 2 2 2 2 2 1 2 2 2 ...
 $ V36 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 3 3 2 2 1 2 2 ...
 $ V37 : Factor w/ 5 levels "?/?","A/A","A/T",..: 4 5 5 5 5 5 5 5 5 5 ...
 $ V38 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 1 1 1 1 1 1 1 1 1 ...
 $ V39 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 4 4 4 4 4 4 4 1 4 ...
 $ V40 : Factor w/ 5 levels "?/?","C/C","C/G",..: 5 2 2 2 2 2 2 2 2 2 ...
 $ V41 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 2 2 2 1 2 2 2 ...
 $ V42 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 5 5 5 5 5 5 5 5 5 ...
 $ V43 : Factor w/ 5 levels "?/?","A/A","A/T",..: 4 1 1 1 1 1 1 1 1 1 ...
 $ V44 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 2 2 2 1 2 2 2 ...
 $ V45 : Factor w/ 5 levels "?/?","A/A","A/T",..: 4 1 1 1 1 1 1 1 1 1 ...
 $ V46 : Factor w/ 5 levels "?/?","A/A","A/T",..: 4 1 5 1 1 5 5 5 5 5 ...
 $ V47 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V48 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 5 5 5 5 5 5 5 5 5 ...
 $ V49 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 4 4 4 4 4 4 4 4 4 ...
 $ V50 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 2 2 2 1 2 2 2 ...
 $ V51 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 1 1 5 5 1 1 1 1 1 ...
 $ V52 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 2 2 1 1 2 2 2 2 2 ...
 $ V53 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 2 2 2 2 2 2 2 2 2 ...
 $ V54 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 1 1 1 1 1 1 1 1 1 ...
 $ V55 : Factor w/ 5 levels "?/?","G/G","G/T",..: 4 5 1 5 5 1 1 5 5 5 ...
 $ V56 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 1 1 2 2 1 2 1 1 1 ...
 $ V57 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 1 1 1 1 1 1 1 1 1 ...
 $ V58 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V59 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V60 : Factor w/ 5 levels "?/?","C/C","C/G",..: 5 2 2 2 2 2 2 2 2 2 ...
 $ V61 : Factor w/ 5 levels "?/?","A/A","A/C",..: 5 4 4 4 4 4 4 4 4 4 ...
 $ V62 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 1 2 2 2 2 2 1 2 ...
 $ V63 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 1 5 5 5 1 1 5 1 5 ...
 $ V64 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V65 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 5 5 5 5 5 5 5 5 5 ...
 $ V66 : Factor w/ 5 levels "?/?","G/G","G/T",..: 4 1 1 1 1 1 1 1 1 1 ...
 $ V67 : Factor w/ 5 levels "?/?","C/C","C/G",..: 5 4 4 4 4 4 4 4 4 4 ...
 $ V68 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 2 2 1 1 2 2 2 2 2 ...
 $ V69 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 1 1 5 1 5 1 1 5 1 ...
 $ V70 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 1 2 2 1 1 2 1 1 1 ...
 $ V71 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 1 1 1 1 1 1 1 1 1 ...
 $ V72 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 5 5 5 5 5 1 5 5 5 ...
 $ V73 : Factor w/ 5 levels "?/?","G/G","G/T",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V74 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 1 1 2 1 1 2 1 1 1 ...
 $ V75 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 1 1 1 1 1 1 1 1 1 ...
 $ V76 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 5 1 5 5 5 1 1 1 1 ...
 $ V77 : Factor w/ 5 levels "?/?","A/A","A/C",..: 5 1 2 2 2 1 2 1 2 1 ...
 $ V78 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 2 1 2 2 2 2 1 ...
 $ V79 : Factor w/ 5 levels "?/?","A/A","A/C",..: 5 2 2 2 2 2 2 2 2 2 ...
 $ V80 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 4 4 4 4 1 4 4 4 4 ...
 $ V81 : Factor w/ 5 levels "?/?","C/C","C/G",..: 5 1 1 1 1 1 1 1 1 1 ...
 $ V82 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 1 2 2 2 1 1 1 1 1 ...
 $ V83 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 4 4 4 4 4 4 4 4 4 ...
 $ V84 : Factor w/ 5 levels "?/?","A/A","A/T",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V85 : Factor w/ 5 levels "?/?","A/A","A/T",..: 4 2 2 1 1 2 1 2 2 2 ...
 $ V86 : Factor w/ 5 levels "?/?","A/A","A/T",..: 4 2 1 2 1 2 1 1 3 1 ...
 $ V87 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 5 5 5 5 5 1 5 5 5 ...
 $ V88 : Factor w/ 5 levels "?/?","A/A","A/C",..: 5 4 1 4 4 4 4 4 4 4 ...
 $ V89 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 1 1 2 2 1 1 1 1 1 ...
 $ V90 : Factor w/ 5 levels "?/?","C/C","C/G",..: 5 2 2 2 2 1 2 1 2 1 ...
 $ V91 : Factor w/ 5 levels "?/?","A/A","A/T",..: 4 5 5 5 5 1 5 1 5 1 ...
 $ V92 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 2 2 2 1 2 2 2 ...
 $ V93 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 3 3 3 3 3 3 3 3 3 ...
 $ V94 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 4 4 4 4 4 4 4 4 4 ...
 $ V95 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 4 4 4 4 4 4 4 4 1 ...
 $ V96 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 1 1 1 1 1 1 1 1 1 ...
 $ V97 : Factor w/ 5 levels "?/?","A/A","A/G",..: 5 1 1 1 1 1 2 1 1 1 ...
 $ V98 : Factor w/ 5 levels "?/?","C/C","C/T",..: 4 2 2 2 2 2 2 2 2 2 ...
 $ V99 : Factor w/ 5 levels "?/?","G/G","G/T",..: 4 5 5 5 5 5 3 5 5 5 ...
  [list output truncated]
  
```{r}
str(snp_position)
```
'data.frame':	983 obs. of  15 variables:
 $ SNP_ID              : Factor w/ 983 levels "abph1.20","abph1.22",..: 1 2 3 4 5 6 7 8 9 10 ...
 $ cdv_marker_id       : int  5976 5978 6605 6606 6607 5982 3463 3466 5983 5985 ...
 $ Chromosome          : Factor w/ 12 levels "1","10","2","3",..: 3 3 6 6 6 1 4 4 5 5 ...
 $ Position            : Factor w/ 942 levels "","100227859",..: 624 625 334 333 335 577 396 395 840 842 ...
 $ alt_pos             : Factor w/ 7 levels "","Approximate: 5 prime BLAST start of amplicon plus 150 bases",..: 1 1 1 1 1 1 1 1 1 1 ...
 $ mult_positions      : Factor w/ 18 levels "","Chr1(275076660);Chr8(71078729);",..: 1 1 1 1 1 1 1 1 1 1 ...
 $ amplicon            : Factor w/ 567 levels "abph1","ae1",..: 1 1 2 2 2 3 4 4 5 5 ...
 $ cdv_map_feature.name: Factor w/ 539 levels "AB042260","ae1",..: 1 1 2 2 2 10 443 443 472 472 ...
 $ gene                : Factor w/ 547 levels "abph1","ae1",..: 1 1 2 2 2 10 463 463 472 472 ...
 $ candidate.random    : Factor w/ 2 levels "candidate","random": 1 1 1 1 1 1 1 1 1 1 ...
 $ Genaissance_daa_id  : int  8393 8394 8395 8396 8397 8398 8399 8400 8401 8402 ...
 $ Sequenom_daa_id     : int  10474 10475 10477 10478 10479 10481 10482 10483 10486 10487 ...
 $ count_amplicons     : int  1 0 1 0 0 1 1 0 1 0 ...
 $ count_cmf           : int  1 0 1 0 0 1 0 0 1 0 ...
 $ count_gene          : int  1 0 1 0 0 1 1 0 1 0 ...
 
#Show the column names
```{r} 
names(fang_et_al_genotypes)
```
  [1] "V1"   "V2"   "V3"   "V4"   "V5"   "V6"   "V7"   "V8"   "V9"   "V10"  "V11"  "V12"  "V13"  "V14"  "V15"  "V16"  "V17"  "V18"  "V19"  "V20" 
 [21] "V21"  "V22"  "V23"  "V24"  "V25"  "V26"  "V27"  "V28"  "V29"  "V30"  "V31"  "V32"  "V33"  "V34"  "V35"  "V36"  "V37"  "V38"  "V39"  "V40" 
 [41] "V41"  "V42"  "V43"  "V44"  "V45"  "V46"  "V47"  "V48"  "V49"  "V50"  "V51"  "V52"  "V53"  "V54"  "V55"  "V56"  "V57"  "V58"  "V59"  "V60" 
 [61] "V61"  "V62"  "V63"  "V64"  "V65"  "V66"  "V67"  "V68"  "V69"  "V70"  "V71"  "V72"  "V73"  "V74"  "V75"  "V76"  "V77"  "V78"  "V79"  "V80" 
 [81] "V81"  "V82"  "V83"  "V84"  "V85"  "V86"  "V87"  "V88"  "V89"  "V90"  "V91"  "V92"  "V93"  "V94"  "V95"  "V96"  "V97"  "V98"  "V99"  "V100"
[101] "V101" "V102" "V103" "V104" "V105" "V106" "V107" "V108" "V109" "V110" "V111" "V112" "V113" "V114" "V115" "V116" "V117" "V118" "V119" "V120"
[121] "V121" "V122" "V123" "V124" "V125" "V126" "V127" "V128" "V129" "V130" "V131" "V132" "V133" "V134" "V135" "V136" "V137" "V138" "V139" "V140"
[141] "V141" "V142" "V143" "V144" "V145" "V146" "V147" "V148" "V149" "V150" "V151" "V152" "V153" "V154" "V155" "V156" "V157" "V158" "V159" "V160"
[161] "V161" "V162" "V163" "V164" "V165" "V166" "V167" "V168" "V169" "V170" "V171" "V172" "V173" "V174" "V175" "V176" "V177" "V178" "V179" "V180"
[181] "V181" "V182" "V183" "V184" "V185" "V186" "V187" "V188" "V189" "V190" "V191" "V192" "V193" "V194" "V195" "V196" "V197" "V198" "V199" "V200"
[201] "V201" "V202" "V203" "V204" "V205" "V206" "V207" "V208" "V209" "V210" "V211" "V212" "V213" "V214" "V215" "V216" "V217" "V218" "V219" "V220"
[221] "V221" "V222" "V223" "V224" "V225" "V226" "V227" "V228" "V229" "V230" "V231" "V232" "V233" "V234" "V235" "V236" "V237" "V238" "V239" "V240"
[241] "V241" "V242" "V243" "V244" "V245" "V246" "V247" "V248" "V249" "V250" "V251" "V252" "V253" "V254" "V255" "V256" "V257" "V258" "V259" "V260"
[261] "V261" "V262" "V263" "V264" "V265" "V266" "V267" "V268" "V269" "V270" "V271" "V272" "V273" "V274" "V275" "V276" "V277" "V278" "V279" "V280"
[281] "V281" "V282" "V283" "V284" "V285" "V286" "V287" "V288" "V289" "V290" "V291" "V292" "V293" "V294" "V295" "V296" "V297" "V298" "V299" "V300"
[301] "V301" "V302" "V303" "V304" "V305" "V306" "V307" "V308" "V309" "V310" "V311" "V312" "V313" "V314" "V315" "V316" "V317" "V318" "V319" "V320"
[321] "V321" "V322" "V323" "V324" "V325" "V326" "V327" "V328" "V329" "V330" "V331" "V332" "V333" "V334" "V335" "V336" "V337" "V338" "V339" "V340"
[341] "V341" "V342" "V343" "V344" "V345" "V346" "V347" "V348" "V349" "V350" "V351" "V352" "V353" "V354" "V355" "V356" "V357" "V358" "V359" "V360"
[361] "V361" "V362" "V363" "V364" "V365" "V366" "V367" "V368" "V369" "V370" "V371" "V372" "V373" "V374" "V375" "V376" "V377" "V378" "V379" "V380"
[381] "V381" "V382" "V383" "V384" "V385" "V386" "V387" "V388" "V389" "V390" "V391" "V392" "V393" "V394" "V395" "V396" "V397" "V398" "V399" "V400"
[401] "V401" "V402" "V403" "V404" "V405" "V406" "V407" "V408" "V409" "V410" "V411" "V412" "V413" "V414" "V415" "V416" "V417" "V418" "V419" "V420"
[421] "V421" "V422" "V423" "V424" "V425" "V426" "V427" "V428" "V429" "V430" "V431" "V432" "V433" "V434" "V435" "V436" "V437" "V438" "V439" "V440"
[441] "V441" "V442" "V443" "V444" "V445" "V446" "V447" "V448" "V449" "V450" "V451" "V452" "V453" "V454" "V455" "V456" "V457" "V458" "V459" "V460"
[461] "V461" "V462" "V463" "V464" "V465" "V466" "V467" "V468" "V469" "V470" "V471" "V472" "V473" "V474" "V475" "V476" "V477" "V478" "V479" "V480"
[481] "V481" "V482" "V483" "V484" "V485" "V486" "V487" "V488" "V489" "V490" "V491" "V492" "V493" "V494" "V495" "V496" "V497" "V498" "V499" "V500"
[501] "V501" "V502" "V503" "V504" "V505" "V506" "V507" "V508" "V509" "V510" "V511" "V512" "V513" "V514" "V515" "V516" "V517" "V518" "V519" "V520"
[521] "V521" "V522" "V523" "V524" "V525" "V526" "V527" "V528" "V529" "V530" "V531" "V532" "V533" "V534" "V535" "V536" "V537" "V538" "V539" "V540"
[541] "V541" "V542" "V543" "V544" "V545" "V546" "V547" "V548" "V549" "V550" "V551" "V552" "V553" "V554" "V555" "V556" "V557" "V558" "V559" "V560"
[561] "V561" "V562" "V563" "V564" "V565" "V566" "V567" "V568" "V569" "V570" "V571" "V572" "V573" "V574" "V575" "V576" "V577" "V578" "V579" "V580"
[581] "V581" "V582" "V583" "V584" "V585" "V586" "V587" "V588" "V589" "V590" "V591" "V592" "V593" "V594" "V595" "V596" "V597" "V598" "V599" "V600"
[601] "V601" "V602" "V603" "V604" "V605" "V606" "V607" "V608" "V609" "V610" "V611" "V612" "V613" "V614" "V615" "V616" "V617" "V618" "V619" "V620"
[621] "V621" "V622" "V623" "V624" "V625" "V626" "V627" "V628" "V629" "V630" "V631" "V632" "V633" "V634" "V635" "V636" "V637" "V638" "V639" "V640"
[641] "V641" "V642" "V643" "V644" "V645" "V646" "V647" "V648" "V649" "V650" "V651" "V652" "V653" "V654" "V655" "V656" "V657" "V658" "V659" "V660"
[661] "V661" "V662" "V663" "V664" "V665" "V666" "V667" "V668" "V669" "V670" "V671" "V672" "V673" "V674" "V675" "V676" "V677" "V678" "V679" "V680"
[681] "V681" "V682" "V683" "V684" "V685" "V686" "V687" "V688" "V689" "V690" "V691" "V692" "V693" "V694" "V695" "V696" "V697" "V698" "V699" "V700"
[701] "V701" "V702" "V703" "V704" "V705" "V706" "V707" "V708" "V709" "V710" "V711" "V712" "V713" "V714" "V715" "V716" "V717" "V718" "V719" "V720"
[721] "V721" "V722" "V723" "V724" "V725" "V726" "V727" "V728" "V729" "V730" "V731" "V732" "V733" "V734" "V735" "V736" "V737" "V738" "V739" "V740"
[741] "V741" "V742" "V743" "V744" "V745" "V746" "V747" "V748" "V749" "V750" "V751" "V752" "V753" "V754" "V755" "V756" "V757" "V758" "V759" "V760"
[761] "V761" "V762" "V763" "V764" "V765" "V766" "V767" "V768" "V769" "V770" "V771" "V772" "V773" "V774" "V775" "V776" "V777" "V778" "V779" "V780"
[781] "V781" "V782" "V783" "V784" "V785" "V786" "V787" "V788" "V789" "V790" "V791" "V792" "V793" "V794" "V795" "V796" "V797" "V798" "V799" "V800"
[801] "V801" "V802" "V803" "V804" "V805" "V806" "V807" "V808" "V809" "V810" "V811" "V812" "V813" "V814" "V815" "V816" "V817" "V818" "V819" "V820"
[821] "V821" "V822" "V823" "V824" "V825" "V826" "V827" "V828" "V829" "V830" "V831" "V832" "V833" "V834" "V835" "V836" "V837" "V838" "V839" "V840"
[841] "V841" "V842" "V843" "V844" "V845" "V846" "V847" "V848" "V849" "V850" "V851" "V852" "V853" "V854" "V855" "V856" "V857" "V858" "V859" "V860"
[861] "V861" "V862" "V863" "V864" "V865" "V866" "V867" "V868" "V869" "V870" "V871" "V872" "V873" "V874" "V875" "V876" "V877" "V878" "V879" "V880"
[881] "V881" "V882" "V883" "V884" "V885" "V886" "V887" "V888" "V889" "V890" "V891" "V892" "V893" "V894" "V895" "V896" "V897" "V898" "V899" "V900"
[901] "V901" "V902" "V903" "V904" "V905" "V906" "V907" "V908" "V909" "V910" "V911" "V912" "V913" "V914" "V915" "V916" "V917" "V918" "V919" "V920"
[921] "V921" "V922" "V923" "V924" "V925" "V926" "V927" "V928" "V929" "V930" "V931" "V932" "V933" "V934" "V935" "V936" "V937" "V938" "V939" "V940"
[941] "V941" "V942" "V943" "V944" "V945" "V946" "V947" "V948" "V949" "V950" "V951" "V952" "V953" "V954" "V955" "V956" "V957" "V958" "V959" "V960"
[961] "V961" "V962" "V963" "V964" "V965" "V966" "V967" "V968" "V969" "V970" "V971" "V972" "V973" "V974" "V975" "V976" "V977" "V978" "V979" "V980"
[981] "V981" "V982" "V983" "V984" "V985" "V986"
```{r}
names(snp_position)
```
 [1] "SNP_ID"               "cdv_marker_id"        "Chromosome"           "Position"             "alt_pos"              "mult_positions"      
 [7] "amplicon"             "cdv_map_feature.name" "gene"                 "candidate.random"     "Genaissance_daa_id"   "Sequenom_daa_id"     
[13] "count_amplicons"      "count_cmf"            "count_gene"        


##Data Processing

#convert file type list into data frame so I can manipulate easier
```{r}
fang <- read.delim("fang_et_al_genotypes.txt")
snp <- read.delim("snp_position.txt")

fang_df <- as.data.frame.matrix(fang) 
snp_df <- as.data.frame.matrix(snp) 
```

#parced out ZMMIL, ZMMLR, and ZMMMR for maize file
#parced out ZMPBA, ZMPIL, ZMPJA for teosinte file 
#transposed files
#used https://medium.com/coinmonks/merging-multiple-dataframes-in-r-72629c4632a3 as reference 
#combine maize data and snp data 
#combine teosinte data and snp data
#used https://stackoverflow.com/questions/52505923/error-in-bind-rows-x-id-argument-1-must-have-names for help
```{r}
maize1 <- data.frame(filter(fang_df, Group == "ZMMIL"))
maize2 <- data.frame(filter(fang_df, Group == "ZMMLR"))
maize3 <- data.frame(filter(fang_df, Group == "ZMMMR"))

maizecat <- do.call("rbind", list(maize1, maize2, maize3))

teosinte1 <- data.frame(filter(fang_df, Group == "ZMPBA"))
teosinte2 <- data.frame(filter(fang_df, Group == "ZMPIL"))
teosinte3 <- data.frame(filter(fang_df, Group == "ZMPJA"))

teosintecat <- do.call("rbind", list(teosinte1, teosinte2, teosinte3))

tmaizecat <- data.frame(t(maizecat))
tteosintecat <- data.frame(t(teosintecat))

noheadm <- tail(tmaizecat, -4)
ordernoheadm <- noheadm[order(noheadm[,1])]

noheadt <- tail(tteosintecat, -4)

noheadsnp <- tail(snp_df, -2)
snpnoheadm <- noheadsnp[order(noheadsnp[,1])]

maizesnp <- data.frame(merge(ordernoheadm, noheadsnp))

work <- tmaizecat[order(tmaizecat[,1])]
workwork <- snp_df[sort(snp_df[,1])]

maizesnp <- data.frame(merge(work, snp_df))
tesintesnp <- data.frame(dplyr::bind_rows(tteosintecat, snp_df))

write.csv(ordernoheadm, file="orderednoheadm.csv")
write.csv(noheadsnp, file="noheadsnp.csv")
write.csv(noheadt, file="noheadt.csv")

```

#My computer kept hitting the error 'Error: vector memory exhausted (limit reached?)' and could not resolve
#computer didnt have enough power 
#borrowed Colleen Yanarella's Windows Laptop (Intel(R) Core(TM) i5-7200U CPU @2.5GHz 2.71 GHz) 64 bit OS 
#see attached file Brianna_more_power.R 
#see code also below

#Read in input
noheadsnp <- read.csv(file = 'noheadsnp.csv')
orderednoheadm <- read.csv(file = 'orderednoheadm.csv')
noheadt <- read.csv(file = 'noheadt.csv')

#Remove weird csv column and rename header
noheadsnpnew <- subset(noheadsnp, select = -X)
colnames(orderednoheadm)[1] <- "SNP_ID"
colnames(noheadt)[1] <- "SNP_ID"

#Merge step
maizesnp <- data.frame(merge(noheadsnpnew, orderednoheadm, by="SNP_ID"))
teosintesnp <- data.frame(merge(noheadsnpnew, noheadt, by="SNP_ID"))

#Change order SNP_ID, Chromosome, Position, ...
maizesnpmerge <-  maizesnp[, c(1, 3, 4, 2, 5:(length(maizesnp)))]
teosintesnpmerge <-  teosintesnp[, c(1, 3, 4, 2, 5:(length(teosintesnp)))]

#Write files to go back to my wimpy device
write.csv(maizesnpmerge, file = "maizesnpmerge.csv")
write.csv(teosintesnpmerge, file = "teosintesnpmerge.csv")


#ordered data by Chromosome
```{r}
maizesnpmerge <- data.frame(read.csv(file = 'maizesnpmerge.csv'))
teosintesnpmerge <- data.frame(read.csv(file = 'teosintesnpmerge.csv'))

maizesnpmerge <- subset(maizesnpmerge, select = -X)
teosintesnpmerge <- subset(teosintesnpmerge, select = -X)

maizesnpmerge <- maizesnpmerge[order(maizesnpmerge$Chromosome),]
teosintesnpmerge <- teosintesnpmerge[order(teosintesnpmerge$Chromosome),]

```


#seperate all the maize chromosomes in order 
```{r}
maizechr1 <- data.frame(filter(maizesnpmerge, Chromosome == 1))
maizechr1$Position <- as.numeric(as.character(maizechr1$Position))
maizechr1 <- maizechr1[order(maizechr1$Position),]
write.table(maizechr1, "maizechr1.txt", sep = "\t")

maizechr2 <- data.frame(filter(maizesnpmerge, Chromosome == 2))
maizechr2$Position <- as.numeric(as.character(maizechr2$Position))
maizechr2 <- maizechr2[order(maizechr2$Position),]
write.table(maizechr2, "maizechr2.txt", sep = "\t")

maizechr3 <- data.frame(filter(maizesnpmerge, Chromosome == 3))
maizechr3$Position <- as.numeric(as.character(maizechr3$Position))
maizechr3 <- maizechr3[order(maizechr3$Position),]
write.table(maizechr3, "maizechr3.txt", sep = "\t")

maizechr4 <- data.frame(filter(maizesnpmerge, Chromosome == 4))
maizechr4$Position <- as.numeric(as.character(maizechr4$Position))
maizechr4 <- maizechr4[order(maizechr4$Position),]
write.table(maizechr4, "maizechr4.txt", sep = "\t")

maizechr5 <- data.frame(filter(maizesnpmerge, Chromosome == 5))
maizechr5$Position <- as.numeric(as.character(maizechr5$Position))
maizechr5 <- maizechr5[order(maizechr5$Position),]
write.table(maizechr5, "maizechr5.txt", sep = "\t")

maizechr6 <- data.frame(filter(maizesnpmerge, Chromosome == 6))
maizechr6$Position <- as.numeric(as.character(maizechr6$Position))
maizechr6 <- maizechr6[order(maizechr6$Position),]
write.table(maizechr6, "maizechr6.txt", sep = "\t")

maizechr7 <- data.frame(filter(maizesnpmerge, Chromosome == 7))
maizechr7$Position <- as.numeric(as.character(maizechr7$Position))
maizechr7 <- maizechr7[order(maizechr7$Position),]
write.table(maizechr7, "maizechr7.txt", sep = "\t")

maizechr8 <- data.frame(filter(maizesnpmerge, Chromosome == 8))
maizechr8$Position <- as.numeric(as.character(maizechr8$Position))
maizechr8 <- maizechr8[order(maizechr8$Position),]
write.table(maizechr8, "maizechr8.txt", sep = "\t")

maizechr9 <- data.frame(filter(maizesnpmerge, Chromosome == 9))
maizechr9$Position <- as.numeric(as.character(maizechr9$Position))
maizechr9 <- maizechr9[order(maizechr9$Position),]
write.table(maizechr9, "maizechr9.txt", sep = "\t")

maizechr10 <- data.frame(filter(maizesnpmerge, Chromosome == 10))
maizechr10$Position <- as.numeric(as.character(maizechr10$Position))
maizechr10 <- maizechr10[order(maizechr10$Position),]
write.table(maizechr10, "maizechr10.txt", sep = "\t")

```

#seperate all teosinte Chr in order 
```{r}
teosintechr1 <- data.frame(filter(maizesnpmerge, Chromosome == 1))
teosintechr1$Position <- as.numeric(as.character(teosintechr1$Position))
teosintechr1 <- teosintechr1[order(teosintechr1$Position),]
write.table(teosintechr1, "teosintehr1.txt", sep = "\t")

teosintechr2 <- data.frame(filter(maizesnpmerge, Chromosome == 2))
teosintechr2$Position <- as.numeric(as.character(teosintechr2$Position))
teosintechr2 <- teosintechr2[order(teosintechr2$Position),]
write.table(teosintechr2, "teosintehr2.txt", sep = "\t")

teosintechr3 <- data.frame(filter(maizesnpmerge, Chromosome == 3))
teosintechr3$Position <- as.numeric(as.character(teosintechr3$Position))
teosintechr3 <- teosintechr3[order(teosintechr3$Position),]
write.table(teosintechr3, "teosintehr3.txt", sep = "\t")

teosintechr4 <- data.frame(filter(maizesnpmerge, Chromosome == 4))
teosintechr4$Position <- as.numeric(as.character(teosintechr4$Position))
teosintechr4 <- teosintechr4[order(teosintechr4$Position),]
write.table(teosintechr4, "teosintehr4.txt", sep = "\t")

teosintechr5 <- data.frame(filter(maizesnpmerge, Chromosome == 5))
teosintechr5$Position <- as.numeric(as.character(teosintechr5$Position))
teosintechr5 <- teosintechr5[order(teosintechr5$Position),]
write.table(teosintechr5, "teosintehr5.txt", sep = "\t")

teosintechr6 <- data.frame(filter(maizesnpmerge, Chromosome == 6))
teosintechr6$Position <- as.numeric(as.character(teosintechr6$Position))
teosintechr6 <- teosintechr6[order(teosintechr6$Position),]
write.table(teosintechr1, "teosintehr6.txt", sep = "\t")


teosintechr7 <- data.frame(filter(maizesnpmerge, Chromosome == 7))
teosintechr7$Position <- as.numeric(as.character(teosintechr7$Position))
teosintechr7 <- teosintechr7[order(teosintechr7$Position),]
write.table(teosintechr7, "teosintehr1.txt", sep = "\t")

teosintechr8 <- data.frame(filter(maizesnpmerge, Chromosome == 8))
teosintechr8$Position <- as.numeric(as.character(teosintechr8$Position))
teosintechr8 <- teosintechr8[order(teosintechr8$Position),]
write.table(teosintechr8, "teosintehr8.txt", sep = "\t")

teosintechr9 <- data.frame(filter(maizesnpmerge, Chromosome == 9))
teosintechr9$Position <- as.numeric(as.character(teosintechr9$Position))
teosintechr9 <- teosintechr9[order(teosintechr9$Position),]
write.table(teosintechr9, "teosintehr9.txt", sep = "\t")

teosintechr10 <- data.frame(filter(maizesnpmerge, Chromosome == 10))
teosintechr10$Position <- as.numeric(as.character(teosintechr10$Position))
teosintechr10 <- teosintechr10[order(teosintechr10$Position),]
write.table(teosintechr10, "teosintehr1.txt", sep = "\t")

```



#maize and teosinte rev chr and ? being changed for -
```{r}

mmsnpdash <- maizesnpmerge
tsnpdash <- teosintesnpmerge


maize_rev <- as_tibble(lapply(mmsnpdash, gsub, pattern="?", replacement="-", fixed=TRUE))
teosinte_rev <- as_tibble(lapply(tsnpdash, gsub, pattern="?", replacement="-", fixed=TRUE))
```


#rev maize - file creation
```{r}
maizechr1rev <- data.frame(filter(maize_rev, Chromosome == 1))
maizechr1rev$Position <- as.numeric(as.character(maizechr1rev$Position))
maizechr1rev <- maize_rev[rev(order(maizechr1rev$Position)),]
write.table(maizechr1rev, "maizechr1rev.txt", sep = "\t")

maizechr2rev <- data.frame(filter(maize_rev, Chromosome == 2))
maizechr2rev$Position <- as.numeric(as.character(maizechr2rev$Position))
maizechr2rev <- maize_rev[rev(order(maizechr2rev$Position)),]
write.table(maizechr2rev, "maizechr2rev.txt", sep = "\t")

maizechr3rev <- data.frame(filter(maize_rev, Chromosome == 3))
maizechr3rev$Position <- as.numeric(as.character(maizechr3rev$Position))
maizechr3rev <- maize_rev[rev(order(maizechr3rev$Position)),]
write.table(maizechr3rev, "maizechr3rev.txt", sep = "\t")

maizechr4rev <- data.frame(filter(maize_rev, Chromosome == 4))
maizechr4rev$Position <- as.numeric(as.character(maizechr4rev$Position))
maizechr4rev <- maize_rev[rev(order(maizechr4rev$Position)),]
write.table(maizechr4rev, "maizechr4rev.txt", sep = "\t")

maizechr5rev <- data.frame(filter(maize_rev, Chromosome == 5))
maizechr5rev$Position <- as.numeric(as.character(maizechr5rev$Position))
maizechr5rev <- maize_rev[rev(order(maizechr5rev$Position)),]
write.table(maizechr5rev, "maizechr5rev.txt", sep = "\t")

maizechr6rev <- data.frame(filter(maize_rev, Chromosome == 6))
maizechr6rev$Position <- as.numeric(as.character(maizechr6rev$Position))
maizechr6rev <- maize_rev[rev(order(maizechr6rev$Position)),]
write.table(maizechr6rev, "maizechr6rev.txt", sep = "\t")

maizechr7rev <- data.frame(filter(maize_rev, Chromosome == 7))
maizechr7rev$Position <- as.numeric(as.character(maizechr7rev$Position))
maizechr7rev <- maize_rev[rev(order(maizechr7rev$Position)),]
write.table(maizechr7rev, "maizechr7rev.txt", sep = "\t")

maizechr8rev <- data.frame(filter(maize_rev, Chromosome == 8))
maizechr8rev$Position <- as.numeric(as.character(maizechr8rev$Position))
maizechr8rev <- maize_rev[rev(order(maizechr8rev$Position)),]
write.table(maizechr8rev, "maizechr8rev.txt", sep = "\t")

maizechr9rev <- data.frame(filter(maize_rev, Chromosome == 9))
maizechr9rev$Position <- as.numeric(as.character(maizechr9rev$Position))
maizechr9rev <- maize_rev[rev(order(maizechr9rev$Position)),]
write.table(maizechr9rev, "maizechr9rev.txt", sep = "\t")

maizechr10rev <- data.frame(filter(maize_rev, Chromosome == 10))
maizechr10rev$Position <- as.numeric(as.character(maizechr10rev$Position))
maizechr10rev <- maize_rev[rev(order(maizechr10rev$Position)),]
write.table(maizechr10rev, "maizechr10rev.txt", sep = "\t")

```


#rev - teosinte file creation
```{r}
teosintechr1rev <- data.frame(filter(teosinte_rev, Chromosome == 1))
teosintechr1rev$Position <- as.numeric(as.character(teosintechr1rev$Position))
teosintechr1rev <- maize_rev[rev(order(teosintechr1rev$Position)),]
write.table(teosintechr1rev, "teosintechr1rev.txt", sep = "\t")

teosintechr2rev <- data.frame(filter(teosinte_rev, Chromosome == 2))
teosintechr2rev$Position <- as.numeric(as.character(teosintechr2rev$Position))
teosintechr2rev <- maize_rev[rev(order(teosintechr2rev$Position)),]
write.table(teosintechr2rev, "teosintechr1rev.txt", sep = "\t")

teosintechr3rev <- data.frame(filter(teosinte_rev, Chromosome == 3))
teosintechr3rev$Position <- as.numeric(as.character(teosintechr3rev$Position))
teosintechr3rev <- maize_rev[rev(order(teosintechr3rev$Position)),]
write.table(teosintechr3rev, "teosintechr3rev.txt", sep = "\t")

teosintechr4rev <- data.frame(filter(teosinte_rev, Chromosome == 4))
teosintechr4rev$Position <- as.numeric(as.character(teosintechr4rev$Position))
teosintechr4rev <- maize_rev[rev(order(teosintechr4rev$Position)),]
write.table(teosintechr4rev, "teosintechr4rev.txt", sep = "\t")

teosintechr5rev <- data.frame(filter(teosinte_rev, Chromosome == 5))
teosintechr5rev$Position <- as.numeric(as.character(teosintechr5rev$Position))
teosintechr5rev <- maize_rev[rev(order(teosintechr5rev$Position)),]
write.table(teosintechr5rev, "teosintechr5rev.txt", sep = "\t")

teosintechr6rev <- data.frame(filter(teosinte_rev, Chromosome == 6))
teosintechr6rev$Position <- as.numeric(as.character(teosintechr6rev$Position))
teosintechr6rev <- maize_rev[rev(order(teosintechr6rev$Position)),]
write.table(teosintechr6rev, "teosintechr6rev.txt", sep = "\t")


teosintechr7rev <- data.frame(filter(teosinte_rev, Chromosome == 7))
teosintechr7rev$Position <- as.numeric(as.character(teosintechr7rev$Position))
teosintechr7rev <- maize_rev[rev(order(teosintechr7rev$Position)),]
write.table(teosintechr7rev, "teosintechr7rev.txt", sep = "\t")

teosintechr8rev <- data.frame(filter(teosinte_rev, Chromosome == 8))
teosintechr8rev$Position <- as.numeric(as.character(teosintechr8rev$Position))
teosintechr8rev <- maize_rev[rev(order(teosintechr8rev$Position)),]
write.table(teosintechr8rev, "teosintechr8rev.txt", sep = "\t")

teosintechr9rev <- data.frame(filter(teosinte_rev, Chromosome == 9))
teosintechr9rev$Position <- as.numeric(as.character(teosintechr9rev$Position))
teosintechr9rev <- maize_rev[rev(order(teosintechr9rev$Position)),]
write.table(teosintechr9rev, "teosintechr9rev.txt", sep = "\t")

teosintechr10rev <- data.frame(filter(teosinte_rev, Chromosome == 10))
teosintechr10rev$Position <- as.numeric(as.character(teosintechr10rev$Position))
teosintechr10rev <- maize_rev[rev(order(teosintechr10rev$Position)),]
write.table(teosintechr10rev, "teosintechr10rev.txt", sep = "\t")

```

#Data Visualization 

#Plot SNPs per Chr
```{r}
countmaize <- dplyr::count(maizesnpmerge, Chromosome, sort = TRUE)
colnames(countmaize)[2] <- "SNP_Counts"

ggplot(countmaize, aes(x = Chromosome, y = SNP_Counts)) + geom_bar(stat="identity") + ggtitle("Number of SNPs per Maize Chromosome")

countteosinte <- dplyr::count(teosintesnpmerge, Chromosome, sort = TRUE)
colnames(countteosinte)[2] <- "SNP_Counts"

ggplot(countteosinte, aes(x = Chromosome, y = SNP_Counts)) + geom_bar(stat="identity") + ggtitle("Number of SNPs per Teosinte Chromosome")

```


#Plot distribution of SNPs on Chr
#referenced https://stackoverflow.com/questions/32051184/replacing-the-specific-values-in-columns-of-data-frame-using-gsub-in-r
#i am aware of extra data points being plotted but i am not sure what to do about it 
```{r}
reducedmaize <- maizesnpmerge[, c("Chromosome", "Position")]

reducedmaize$Position <- sub("unknown", "", reducedmaize$Position)

ggplot(reducedmaize, aes(x = Chromosome, y = Position)) + geom_point(stat="identity") + ggtitle("SNP Distribution on Maize Chromosomes")


reducedteosinte <- teosintesnpmerge[, c("Chromosome", "Position")]

reducedteosinte$Position <- sub("unknown", "", reducedteosinte$Position)

ggplot(reducedteosinte, aes(x = Chromosome, y = Position)) + geom_point(stat="identity") + ggtitle("SNP Distribution on Teosinte Chromosomes")

```


#homo or het count
#couldn't figure out automated code went in maunally 
#https://stackoverflow.com/questions/44833947/counting-number-of-specific-strings-in-an-r-data-frame
```{r}

length(maizesnpmerge$X2504)

X2504 <- data.frame(count(maizesnpmerge, X2504))

homoX2504 <- as.numeric(sum(X2504[2,2],X2504[3,2],X2504[6,2], X2504[7,2]))

hetX2504 <- as.numeric(sum(X2504[4,2],X2504[5,2]))

unknownX2504 <- as.numeric(sum(X2504[1,2]))

genotypeX <- c("homo", "het", "unknown")
valueX <- c(homoX2504, hetX2504, unknownX2504)
plotX <- data.frame(genotypeX, valueX)

ggplot(data = plotX, mapping = aes(x=genotypeX, y=valueX)) + geom_col()

```

#wasn't sure what the last part was asking me for